tests/frontend/test_starship.py

```diff
@@ -0,0 +1,486 @@
+            m.setattr(cli, "store_account", mock_store_account.store_account)
+
+            args = MockArgs()
+            args.token = "SomeToken"
+
+            cli.configure(args)
+            assert mock_store_account.kwargs == {"authentication_token": "SomeToken"}
+
+    def test_configuration_wizard(self, monkeypatch):
+        """Tests that if no token was given as a command line argument then
+        configuration takes place using the configuration_wizard function."""
+        with monkeypatch.context() as m:
+            mock_store_account = MockStoreAccount()
+            m.setattr(cli, "configuration_wizard", lambda: cli.create_config()["api"])
+            m.setattr(cli, "store_account", mock_store_account.store_account)
+
+            args = MockArgs()
+            args.token = False
+
+            cli.configure(args)
+            assert mock_store_account.kwargs == EXPECTED_KWARGS
+
+    def test_token_local(self, monkeypatch):
+        """Tests that if a token was given as a command line argument and
+        local configuration was specified then configuration takes place
+        accordingly."""
+        with monkeypatch.context() as m:
+            mock_store_account = MockStoreAccount()
+            m.setattr(cli, "store_account", mock_store_account.store_account)
+
+            args = MockArgs()
+            args.token = "SomeToken"
+            args.local = True
+
+            cli.configure(args)
+            assert mock_store_account.kwargs == {"authentication_token": "SomeToken", "location": "local"}
+
+    def test_configuration_wizard_local(self, monkeypatch):
+        """Tests that if no token was given as a command line argument and
+        local configuration was specified then configuration takes place using
+        the configuration_wizard function."""
+        with monkeypatch.context() as m:
+            mock_store_account = MockStoreAccount()
+            m.setattr(cli, "configuration_wizard", lambda: cli.create_config()["api"])
+            m.setattr(cli, "store_account", mock_store_account.store_account)
+
+            args = MockArgs()
+            args.token = False
+            args.local = True
+
+            cli.configure(args)
+            EXPECTED_KWARGS["location"] = "local"
+
+            assert mock_store_account.kwargs == EXPECTED_KWARGS
+
+class MockSuccessfulConnection:
+    """A Connection class mocking a successful establishment of connection."""
+
+    def ping(self):
+        return True
+
+class MockFailedConnection:
+    """A Connection class mocking a failed establishment of connection."""
+
+    def ping(self):
+        return False
+
+class TestPing:
+    """Tests for the pinging mechanism of the CLI."""
+
+    def test_success(self, monkeypatch, capsys):
+        """Test that pinging was successful."""
+        with monkeypatch.context() as m:
+            m.setattr(cli, "Connection", MockSuccessfulConnection)
+            cli.ping()
+
+        out, _ = capsys.readouterr()
```

Comment:
This is cool, learned something new today :+1:

Reply:
Was new for me as well, @josh146 pointed me to it! :blush: